code/game/objects/items/mountable_frames/frames.dm

```diff
@@ -25,3 +25,16 @@
 			user << "<span class='rose'>[src] cannot be placed in this area.</span>"
 			return
 		return 1
+
+/obj/item/mounted/frame/newscaster
```

Comment:
>all in the parent file

Except that lack of organisation makes vending.dm woefully unreadable and impossible to edit without getting a headache.